### PR TITLE
ENH some more yaml patches

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist

* [X] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/patch_yaml/README.md)).
* [ ] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [ ] Ran `python show_diff.py` and posted the output as part of the PR.
* [ ] Modifications won't affect packages built in the future. 

<!-- Put any other comments or information here --!>

### CODE REVIEW

**Comment by jaimergp** [comment]
I'll try to finish the list today @beckermr

File: recipe/patch_yaml/keyring.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#         if dep_name == "importlib_metadata" and ">=" not in dep:
+#             record["depends"][i] = "importlib_metadata >=3.6"
+if:
+  name: keyring
+  version: "23.6.0"
+  build_number: 0
+then:
+  - replace_depends:
+      old: "importlib_metadata?( *)[!>=]*"
+      new: "importlib_metadata >=3.6"
```
Comment (beckermr) [inline recipe/patch_yaml/keyring.yaml:16]:
```suggestion
      new: "importlib_metadata >=3.6"

```

Comment (jaimergp) [reply recipe/patch_yaml/keyring.yaml]: I'll be committing during the next few hours to this PR, and then I'll run the pre-commit hooks, consider it WIP til I mark it as ready for review. Sorry for the noise and thanks! 🙏

**Comment by jaimergp** [comment]
> Btw, We have to run show diff for all of these to ensure the patches are the same. This is how I've been testing these prs.

Yep, I will ensure the diffs are unchanged. I can't do it before committing because I am on my phone data from a train so downloading the repodata is a bit too much 😬

**Comment by beckermr** [comment]
The ci job runs it so you can copy paste from there.

**Comment by jaimergp** [comment]
And done with most of them. Untested so errors are expected, I'll polish later when I have reliable connection. This was fun 😂

File: recipe/patch_yaml/cfn-lint.yaml
Comment (jaimergp) [inline recipe/patch_yaml/cfn-lint.yaml:1]: This gets an award 🏅 Unsure if we should leave it as Python though 🤷

File: recipe/patch_yaml/depfinder.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#     if pind is not None:
+#         record["depends"][pind] = record["depends"][pind] + ",<3.10"
+if:
+  name: depfinder
+  timestamp_lt: 1659704295850
+  subdir: noarch
+  has_depends: "python *[!<]*"
+then:
+  - add_depends: python <3.10  # FIXME? this uses matchspec merge logic
```
Comment (jaimergp) [inline recipe/patch_yaml/depfinder.yaml:28]: How do you feel about this type of upper bound capping? It _should_ work just as fine given how `conda` merges `MatchSpec` objects (logical AND if compatible).

Comment (beckermr) [reply recipe/patch_yaml/depfinder.yaml:28]: I think this is fine. My plan had been to make an update to _pin_stricter to handle more cases like this. See this draft PR here: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/518/files

File: recipe/patch_yaml/cfn-lint.yaml
Comment (beckermr) [reply recipe/patch_yaml/cfn-lint.yaml:1]: OMG. It is fine as is but man it is long. Thank you!

**Comment by beckermr** [comment]
OK the final diff only removes duplicate entries which is totally fine. 

```diff
================================================================================
================================================================================
noarch
noarch::python-slugify-8.0.1-pyhd8ed1ab_1.conda
-    "awesome-slugify <0",
-    "slugify <0",
================================================================================
================================================================================
linux-64
linux-64::cub-1.16.0-h0800d71_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-64 <0.0.0a0",
linux-64::thrust-1.16.0-h0800d71_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-64 <0.0.0a0",
================================================================================
================================================================================
linux-armv7l
================================================================================
================================================================================
linux-aarch64
linux-aarch64::cub-1.16.0-h1212f3d_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-aarch64 <0.0.0a0",
linux-aarch64::thrust-1.16.0-h1212f3d_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-aarch64 <0.0.0a0",
================================================================================
================================================================================
linux-ppc64le
linux-ppc64le::cub-1.16.0-hc9efd5c_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-ppc64le <0.0.0a0",
linux-ppc64le::thrust-1.16.0-hc9efd5c_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_linux-ppc64le <0.0.0a0",
================================================================================
================================================================================
osx-64
osx-64::thrust-1.16.0-h7e98741_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_osx-64 <0.0.0a0",
================================================================================
================================================================================
osx-arm64
================================================================================
================================================================================
win-32
================================================================================
================================================================================
win-64
win-64::cub-1.16.0-h9631440_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_win-64 <0.0.0a0",
win-64::thrust-1.16.0-h9631440_2.conda
-    "cuda-cccl <0.0.0a0",
-    "cuda-cccl-impl <0.0.0a0",
-    "cuda-cccl_win-64 <0.0.0a0",
```

**Comment by beckermr** [comment]
I double checked the duplicates on anaconda.org for each kind.

File: recipe/patch_yaml/cppyy.yaml [hunk 3]
```diff
@@ -24,8 +24,10 @@ if:
...
+  not_has_depends: libcxx >*,<*
 then:
-  - tighten_depends:
-      name: libcxx
-      max_pin: x
-      upper_bound: "15"
+  - replace_depends:
+      # logic is not the same as above put patch is the same
+      old: libcxx?( *)
+      new: ${old},<15
```
Comment (jaimergp) [inline recipe/patch_yaml/cppyy.yaml:33]: Ohh I was wondering if you could use something like this. Nice!

**Comment by jaimergp** [comment]
Thanks for the fixes, Matt!

**Pull request merged**